### ISSUE TITLE
Issue but maybe not an issue

### ISSUE THREAD

**Issue by ermurenz** [gwojcieszczuk/kms4vsphere#3]
Hi guys,
my english sucks so i apologize in advance.
This is my experience.

I uploaded the kms4vsphere ova on our vCenter VMware with v6.7 an it work perfectly on any aspect.
All clean on both sides (appliance and vcenter).
We tried with a vCenter VMware 7.0.2.00400 and it seems to work fine: encryption and decryption works and vcenter  not show any issue.
But,  the /var/lib/lxd/storage-pools/default/containers/KMS/rootfs/var/log/pykmip/server.log
of the PyKMIP side, show this type of log:

2022-02-13 02:07:36,575 - kmip.server.engine - INFO - Processing operation: GetAttributes
2022-02-13 02:07:36,583 - kmip.server.session.00000040 - WARNING - Failure parsing request message.
2022-02-13 02:07:36,583 - kmip.server.session.00000040 - ERROR -
Traceback (most recent call last):
  File "/usr/local/lib/python3.8/dist-packages/PyKMIP-0.11.0.dev1-py3.8.egg/kmip/services/server/session.py", line 167, in _handle_message_loop
    request.read(request_data, kmip_version=kmip_version)
  File "/usr/local/lib/python3.8/dist-packages/PyKMIP-0.11.0.dev1-py3.8.egg/kmip/core/messages/messages.py", line 485, in read
    batch_item.read(istream, kmip_version=kmip_version)
  File "/usr/local/lib/python3.8/dist-packages/PyKMIP-0.11.0.dev1-py3.8.egg/kmip/core/messages/messages.py", line 307, in read
    self.request_payload = self.payload_factory.create(
  File "/usr/local/lib/python3.8/dist-packages/PyKMIP-0.11.0.dev1-py3.8.egg/kmip/core/factories/payloads/__init__.py", line 48, in create
    return self._create_add_attribute_payload()
  File "/usr/local/lib/python3.8/dist-packages/PyKMIP-0.11.0.dev1-py3.8.egg/kmip/core/factories/payloads/__init__.py", line 147, in _create_add_attribute_payload
    raise NotImplementedError()
NotImplementedError
2022-02-13 02:07:36,585 - kmip.server.session.00000040 - INFO - Stopping session: 00000040

This thing happens only when there is the finalization of the encryption procedure of the virtual machines:
Literally on the last second , but every time.
No problem on decryptions.
The appliance does its duty, everything seems to work correctly, the vms are encrypted and decrypted without problems on the vmware side.
All "actions" on the encrypted vms goes fine.
No error showed.
In short, no problem except that particular log.
Doesn't seem the case but, could it be a symptom of some worse stuff?
I've tried multiple upload of ova with the same results.
I also tried manual installation (with ubuntu  iso, PyKMIP and setup.py, etc)
but the same thing happened.
I hope thit is the right section for this type of question.If not, i apologise.

any advise is appreciate.

**Comment by gwojcieszczuk** [comment]
Hi there,
Don't have vcenter 7.x at the moment to test it. I don't see this error when using with vcenter 6.7. Based on "NotImplementedError", it seems benign. Maybe vcenter 7.x is requesting some information (optional) that vcenter 6.7 didn't??? Keep in mind also, that ESXi7 can now persistently cache encryption keys (survive reboot); ESXi 6.x was requesting encryption keys from KMS on every boot.

**Comment by ermurenz** [comment]
> Hi there, Don't have vcenter 7.x at the moment to test it. I don't see this error when using with vcenter 6.7. Based on "NotImplementedError", it seems benign. Maybe vcenter 7.x is requesting some information (optional) that vcenter 6.7 didn't??? Keep in mind also, that ESXi7 can now persistently cache encryption keys (survive reboot); ESXi 6.x was requesting encryption keys from KMS on every boot.

Hi! Thank for your reponse.
I like this project and the fact that after a reboot of the vcenter , there is no need to do anything. Everything works.
Yes, as i said I haven't seen any errors of this type on 6.7 either  (and again, no on "vSphere client (html5)" of vcenter 7.x ).
I am comforted to understand that from your point of view it looks like a "benign" alert.
You know if there will be new releases of kms4vsphere (and PyKMIP i guess) in the future?

tnks in advance
